bokehjs/src/lib/models/widgets/tables/data_table.ts

```diff
@@ -247,19 +288,39 @@ export class DataTableView extends WidgetView {
       reorderable = false
     }
 
+    let frozenRow = -1
+    let frozenBottom = false
+    const frozenColumn = this.model.frozen_columns == null ? -1 : this.model.frozen_columns-1
+    if (this.model.frozen_rows != null) {
+      frozenBottom = this.model.frozen_rows < 0 ? true : false
+      frozenRow = Math.abs(this.model.frozen_rows)
+    }
+
     const options = {
       enableCellNavigation: this.model.selectable !== false,
       enableColumnReorder: reorderable,
-      forceFitColumns: this.model.fit_columns, // TODO: update to autosizeColsMode
+      autosizeColsMode: this.autosize,
       multiColumnSort: this.model.sortable,
       editable: this.model.editable,
-      autoEdit: false,
+      autoEdit: this.model.auto_edit,
+      autoHeight: false,
       rowHeight: this.model.row_height,
+      frozenColumn,
+      frozenRow,
+      frozenBottom,
     }
 
     this.data = new TableDataProvider(this.model.source, this.model.view)
     this.grid = new SlickGrid(this.el, this.data, columns, options)
 
+    if (this.autosize == AutosizeModes.fit_viewport) {
+      this.grid.autosizeColumns()
+      let width = 0
+      for (const column of columns)
+        width += (column as any).width
```

Comment:
This `any` seems unnecessary, as `TableColumn` has such property.

Reply:
`column` here is the return value of `TableColumn.toColumn()`, probably needs better typing since that's just typed as:

```python
import {Column} from "@bokeh/slickgrid"
export type ColumnType = Column<Item> & {model?: CellEditor}
```

Reply:
Right, different structure, but it still has `width?: number`, so `width += column.width ?? 0` should work fine.